linear_model.py

```diff
@@ -0,0 +1,118 @@
+
+import os
+import sys
+
+import tensorflow as tf
+
+class LinearModel:
+
+    def __init__(self, input_size, output_size, checkpoint=None):
+        self.input_size = input_size
+        self.output_size = output_size
+
+        self.sess = tf.Session()
+        self._build()
+        self.saver = tf.train.Saver()
+
+        if checkpoint is not None:
+            self.restore_model(checkpoint)
+        else:
+            self.sess.run(tf.global_variables_initializer())
+
+        self.iter_number = 0
+
+    ## Public methods
+
+    def predict(self, state):
+        """
+        state: B x I
+        """
+        output = self.sess.run(self.out, feed_dict={self.input_state: state})
+        return output
+
+    def train_reinforce(self, state, pred, action, var, reward):
+        """
+        B = batch size
+        I = input size
+        A = action dimension
+
+        state: B x I    input state being trained
+        pred: B x A     output from model.predict(state)
+        action: B x A   sampled action using mean from pred
+        var: B x A      variance used for sampling action (MUST BE POSTIIVE)
+        reward: B x 1   reward from environment for the performed action
+
+        inputs expected to be numpy arrays
+        """
+
+        # TODO: for actor-critic, pred should be predicted from a separate critic network instead
+
+        target = (action - pred) * reward / var
```

Comment:
Yeah, I'm missing a 2 in the denominator, but the rest of it is actually correct. If I'm interpreting things correctly, the pred, is the predicted mean (it's a B x 2 vector) which includes both the predicted mean accel_x and mean accel_y. Likewise, action also includes both accel_x and accel_y already (see the doc string).